src/main/java/project/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,83 @@
+package project.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import project.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import project.bookstore.dto.category.CategoryDto;
+import project.bookstore.dto.category.CategoryRequestDto;
+import project.bookstore.service.BookService;
+import project.bookstore.service.CategoryService;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+@Tag(name = "Categories management", description = "Endpoints for managing categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
I suggest you to use the hasAnyRole method and specify two roles ('ROLE_USER', 'ROLE_ADMIN') instead of using hasRole.
```suggestion
    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
```